### ISSUE TITLE
Dimension 2 in both shapes must be equal, but are 1 and 3 for 'Assign' (op: 'Assign') with input shapes: [3,3,1,32], [3,3,3,32].

### ISSUE THREAD

**Issue by hnsywangxin** [fchollet/keras#6237]
I used the Pre-training model which is Xception,but hint the dimension is inconsistent,my dataset is the fer 2013 which  consists of 48x48 pixel grayscale images of faces,I have change 3 to 1,but there are still the issue,thank  you  very much!

**Comment by patyork** [comment]
From the [documentation](https://keras.io/applications/#xception), emphasis added:

>input_shape: optional shape tuple, only to be specified if include_top is False (otherwise the input shape has to be  (299, 299, 3). It should have exactly 3 inputs channels, and **width and height should be no smaller than 71**. E.g.  (150, 150, 3) would be one valid value.

**Comment by hnsywangxin** [comment]
I have seen it ,it means I should input RGB image,but my dataset is grayscale which only 1 channels,How to change gray image to RGB image with python? thank you @patyork

**Comment by patyork** [comment]
You can do `np.stack` or something similar to copy the data and make 2 extra fake channels. That doesn't change the fact that Xception expects 71x71 pixel (or larger) images, and will still break if you feed it 48x48.

**Comment by hnsywangxin** [comment]
that mean my dataset is not suitable for this model?

**Comment by patyork** [comment]
Not really. All of the Keras image applications expect RGB images, but grayscale images (if stacked) should work OK.

VGG16 and VGG19 require 48x48 (or larger) images, so they may be better suited. You could also try to UpSize your dataset (to 71x71 for Xception), but this may do more harm than good. Padding your images with a border might work as well. You'd have to test all of these things to see which one works best for your dataset.

**Comment by hnsywangxin** [comment]
thank you @patyork
